packages/@react-stately/table/src/TableCollection.ts

```diff
@@ -216,6 +217,30 @@ export class TableCollection<T> extends GridCollection<T> {
     for (let node of nodes) {
       visit(node);
     }
+
+    if (Array.from(nodes).some(node => node.props?.allowsResizing)) {
+      // add a hidden buffer column to the end of the table
+      // this will be used to allow for smooth scrolling when
+      // resizing columns while a horizontal scroll bar is present
+      // (also lovingly known as the "spooky column")
+      let resizeBufferColumn: GridNode<T> = {
```

Comment:
@marshallpete could you take care of this one? ☝️

Reply:
If the content of the table > table width, you get a horizontal scroll bar.
If you then scroll to the end of the content horizontally, and then try to shrink one of the columns by resizing it, the table content width gets smaller, changing how big the horizontal scroll bar is.
This causes some stuff to snap around and breaks the resize behavior.

To fix this, we add a "spooky column" to the end of the table. The width of this column defaults to 0. If you try and shrink a column and the width of the table contents > table width, then the "spooky column" will grow to take up the difference so that the total table content width remains constant while you are resizing. Once you finish resizing, the  "spooky column" snaps back to 0.

Reply:
https://user-images.githubusercontent.com/40001449/157966303-62256761-0f3b-4784-8a33-12b42b33aba7.mov

Reply:
Added comments to the code as well: [77c0b56](https://github.com/adobe/react-spectrum/pull/2883/commits/77c0b564176687c1574f44b6695e4167a3a451a7)